### PR TITLE
Error corrections

### PR DESCRIPTION
**Pull request by iOiurson**
This PR fixes most of remaining errors:
- **option added to ignore hidden cells** when reading .XLSX files, which caused most of the remaining Errors on our related project. This option must be used wisely, since hidden cells won't be part of the graph.
- better (but not perfect yet) **management of `ExcelError`s in `excellib` functions**
- **check if a `Range` has keys** before trying to access its `addresses` (=> maybe find a better way to handle Range that are not initialized correctly)

Also, added a `debug` check for debug prints.

This PR leads us from 1200+ to **107 errors** on the main calculation path of our main project (3 different error types).
Bad evaluations: +22 => still about 1300 to handle.

Total would be (if run on all files):
```
After evalling 13963 files: Good: 12499, Bad: 1357, Errors: 107 (3 different)
```

### CODE REVIEW

**Comment by iOiurson** [comment]
FYI, finer grain error analysis shows 5 different error types, 2 of them combinated representing 103 files.

**Comment by iOiurson** [comment]
With the latest commits, we're down to 67 errors (5 different), and one error represents 63 files:
````
Problem evalling: maximum recursion depth exceeded while calling a Python object for Calculations!AU297, RangeCore.apply('multiply',iferror(RangeCore.apply('add',RangeCore.apply('add',self.eval_ref(\"Calculations!AT297\", ref = (297, 'AU')),self.eval_ref(\"Calculations!AT337\", ref = (297, 'AU')),(297, 'AU')),self.eval_ref(\"Depreciation!AT65\", ref = (297, 'AU')),(297, 'AU')),0),self.eval_ref('CA_OpTrig', ref = (297, 'AU')),(297, 'AU'))
```

**Comment by vallettea** [comment]
what a day ! 👍

**Comment by SergeBredin** [comment]
💣💣💣💣💣💣💣💣

**Comment by iOiurson** [comment]
This should leave us with ZERO errors.
Be careful though, you need to provide enough stack depth for calculations, if you want to avoid recursion errors. See `eval.py`.
You can merge.

**Comment by iOiurson** [comment]
Also, latest commit (https://github.com/iOiurson/koala/commit/acb0333b7d75fbd58967bc4d8ea077edc78bb003) doesn't seem to have an effect on kerrigan, but works on my machine.
We're going to have to investigate it, but later.

File: eval.py [hunk 1]
```diff
@@ -20,38 +22,41 @@
...
+    sp = Spreadsheet.load2(file.replace("xlsx", "gzip").replace(folder, "graphs"))
     print "___Timing___ Graph read in %s" % (str(datetime.now() - startTime))
 
     # import cProfile
     # cProfile.run('Spreadsheet.load2(file.replace("xlsx", "txt"))', 'stats')
 
-    sys.setrecursionlimit(10000)
+    sys.setrecursionlimit(30000)
+    limit = 67104768 # maximum stack limit on my machine => use 'ulimit -Ha' on a shell terminal
+    resource.setrlimit(resource.RLIMIT_STACK, (limit, limit))
```
Comment (vallettea) [inline eval.py]: is this machine specific ?

Comment (iOiurson) [inline eval.py]: Yes, the `limit` is machine specific. You can find it by running `ulimit -Ha`.
With this particular value (which is my max value), it works on my machine, but not on kerrigan, which is weird. But kerrigan has unlimited limit on this matter, so we have options. We'll have to investigate more.

**Comment by vallettea** [comment]
is it normal that all tests are commented ?

**Comment by iOiurson** [comment]
No, i forgot to uncomment, but it's no big deal, i will change it back later.

**Comment by vallettea** [comment]
looked hard, nice job.

**Pull request merged**